### PR TITLE
Implement MutableArithmetics

### PR DESCRIPTION
**Pull request by blegat**
- [x] Add tests
- [x] Add bounds to MA in Project.tom
- [x] Release MA

cc @tweisser @Peiffap @saschatimme

### CODE REVIEW

**Comment by blegat** [comment]
Looks ready from my side, any objection ?

**Comment by mlubin** [comment]
I'd like to take a look before it's merged.

File: src/Utilities/mutable_arithmetics.jl [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+MA.mutability(::Type{<:TypedLike}) = MA.IsMutable()
+
+function MA.mutable_copy(func::MOI.ScalarAffineFunction)
+    terms = [MOI.ScalarAffineTerm(MA.copy_if_mutable(t.coefficient), t.variable_index) for t in func.terms]
+    return MOI.ScalarAffineFunction(terms, MA.copy_if_mutable(func.constant))
+end
+function MA.mutable_copy(func::MOI.ScalarQuadraticFunction)
+    affine_terms = [MOI.ScalarAffineTerm(MA.copy_if_mutable(t.coefficient), t.variable_index) for t in func.affine_terms]
+    quadratic_terms = [MOI.ScalarQuadraticTerm(MA.copy_if_mutable(t.coefficient), t.variable_index_1, t.variable_index_2) for t in func.quadratic_terms]
```
Comment (mlubin) [inline src/Utilities/mutable_arithmetics.jl]: This is stretching the character limit

**Comment by blegat** [comment]
All lines in the diff should be covered now, the remaining lines claimed to be uncovered by codecov are in fact covered but codecov does not see it as they are inlined.

**Pull request merged**